xbmc/windowing/gbm/DRMAtomic.cpp

```diff
@@ -22,6 +23,24 @@
 #include <unistd.h>
 
 using namespace KODI::WINDOWING::GBM;
+bool CDRMAtomic::SetScalingFilter(struct drm_object *object, const char *name, const char *type) {
+  uint32_t w_scale_factor;
+  uint32_t h_scale_factor;
+  for (uint32_t i = 0; i < object->props->count_props; i++)
+      if (!strcmp(object->props_info[i]->name, name))
```

Comment:
please use `CDRMUtils::SupportsProperty`. You may also want to use `CDRMUtils::SupportsPropertyAndValue`

Reply:
CDRMUtils::SupportsProperty  & CDRMUtils::SupportsPropertyAndValue check particular property & value passed exists or not, but in this particular case we are traversing through Property and when we find the property (SCALING_FILTER) and then traverse through its supported types ("Nearest Neighbor", "Default" ..etc) then we set type ("Nearest Neighbor") value.
But wherever needs to check the property existence, will use CDRMUtils::SupportsProperty  & CDRMUtils::SupportsPropertyAndValue.
Ex: 
if (SupportsProperty(m_gui_plane, "SCALING_FILTER"))
  {
    const std::shared_ptr<CSettings> settings =
        CServiceBroker::GetSettingsComponent()->GetSettings();
    settings->GetSetting(SETTING_VIDEOSCREEN_HW_SCALING_FILTER)->SetVisible(true);
  }

Reply:
I disagree, you could simply do
```
if (SupportsPropertyAndValue(m_gui_plane, "SCALING_FILTER", "Nearest Neighbor"))
  AddProperty(m_gui_plane, "SCALING_FILTER", "Nearest Neighbor")

```

You are asking for a specific value ("Nearest Neighbor") anyways, so there is no reason to traverse all the values. You could traverse all the values if you wanted to create a settingsfilter that will populate based on the enum values of "SCALING_FILTER", but that is only needed if you want to use other methods than just "Nearest Neighbor" and "Default".

Reply:
Let me know if this is ok, I will incorporate.
I validated it and it works fine.
-----------------------------------------------
enum drm_scaling_filter
{
  DRM_SCALING_FILTER_DEFAULT,
  DRM_SCALING_FILTER_NEAREST_NEIGHBOR,
};
int32_t CDRMAtomic::GetScalingFilterType(const char* type)
{
  if (!strcmp(type, "Nearest Neighbor"))
    return DRM_SCALING_FILTER_NEAREST_NEIGHBOR;
  else
    return DRM_SCALING_FILTER_DEFAULT;
}
bool CDRMAtomic::SetScalingFilter(struct drm_object* object, const char* name, const char* type)
{
  uint32_t iType = GetScalingFilterType(type);
  if (SupportsPropertyAndValue(object, name, iType))
    if (AddProperty(object, name, iType))
    {
      CLog::Log(LOGDEBUG, "CDRMAtomic::{}, Window size: {}x{}", __FUNCTION__, m_width, m_height);
     < . . . >
     return true;
    }
  return false;
}